File: qc-domain/src/main/java/com/eager/questioncloud/domain/post/repository/PostCommentRepository.java
package com.eager.questioncloud.domain.post.repository;

import com.eager.questioncloud.common.PagingInformation;
import com.eager.questioncloud.domain.post.dto.PostCommentDetail;
import com.eager.questioncloud.domain.post.model.PostComment;
import java.util.List;

public interface PostCommentRepository {
    PostComment save(PostComment postComment);

    PostComment findByIdAndWriterId(Long commentId, Long userId);

    List<PostCommentDetail> getPostCommentDetails(Long postId, Long userId, PagingInformation pagingInformation);

    void delete(PostComment postComment);

    int count(Long postId);
}

File: qc-domain/qc-post/src/main/kotlin/com/eager/questioncloud/post/infrastructure/repository/PostCommentRepository.kt
package com.eager.questioncloud.post.infrastructure.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.post.domain.PostComment

interface PostCommentRepository {
    fun save(postComment: PostComment): PostComment

    fun findByIdAndWriterId(commentId: Long, userId: Long): PostComment

    fun findByPostIdWithPagination(
        postId: Long,
        pagingInformation: PagingInformation
    ): List<PostComment>

    fun delete(postComment: PostComment)

    fun count(postId: Long): Int
}
